src/OpenTK.GraphicsLibraryFramework/Interfaces/IGLFW.cs

```diff
@@ -0,0 +1,72 @@
+﻿//
+// IGLFW.cs
+//
+// Copyright (C) 2018 OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+using System;
+using AdvancedDLSupport;
+
+namespace OpenToolkit.GraphicsLibraryFramework
+{
+    /// <summary>
+    ///     Defines the public interface of the GLFW 3.2 library
+    /// </summary>
+    [NativeSymbols(Prefix = "glfw")]
+    internal interface IGLFW
+    {
+        bool Init();
+        void Terminate();
+        unsafe Window* CreateWindow(int width, int height, string title, Monitor* monitor, Window* share);
+        unsafe void DestroyWindow(Window* window);
+        unsafe void FocusWindow(Window* window);
+        unsafe string GetClipboardString(Window* window);
+        unsafe void GetFramebufferSize(Window* window, out int width, out int height);
+        unsafe InputModeValue GetInputMode(Window* window, InputMode mode);
+        unsafe Monitor* GetPrimaryMonitor();
+        unsafe VideoMode* GetVideoMode(Monitor* monitor);
+        unsafe int GetWindowAttrib(Window* window, int attribute);
+        unsafe void GetWindowSize(Window* window, out int width, out int height);
+        unsafe void GetWindowPos(Window* window, out int x, out int y);
+        unsafe Monitor* GetWindowMonitor(Window* window);
+        unsafe void HideWindow(Window* window);
+        unsafe void IconifyWindow(Window* window);
+        unsafe void MakeContextCurrent(Window* window);
+        unsafe void MaximizeWindow(Window* window);
+        void PollEvents();
+        void PostEmptyEvent();
+        unsafe void RestoreWindow(Window* window);
+        unsafe void SetCharCallback(Window* window, Action<Window, int> callback); //TODO: Make enum for int param in callback
```

Comment:
As I'm currently doing a little bit of XML documentation, because I guess that it is something nobody wants to do :P - I noticed that in the actual documentation of these callback setting functions the return values should actually be the previous callback value and therefore the return type of void is wrong.
Question is how would the other guys here want to have that resolved? Perhaps make a specific delegate for each callback type would make sense as well and then let it return that. I could do it, I just wanted to confirm what a desired approach would be.
Or can AdvancedDL even work with delegate return values?

Reply:
cc: @devvoid

Reply:
I don't really understand delegates or callbacks well enough to give an answer here. I think Dae would have to handle this.

Reply:
If I had to fathom a guess, it probably returns an IntPtr which serves as a function pointer to the delegate that was mapped. Lets go deeper, @Nihlus what kind of effort is it to map that and it my assumption even close?

Reply:
Well yes it is an IntPtr pointing to a native function pointer, and normally should be callable with either calli(IL code), or by converting it to a delegate with Marshal.GetDelegateForPtr(whereas calli has a much better performance, even if we would create a delegate which uses calli in itself).

I would care what was previously mapped, as if someone else uses the API, I don't need to think much about it, but can just replace his delegate with mine, while inside my function calling his delegate, so that the old delegate still gets the information and is not lost in the void...